pkg/register/component.go

```diff
@@ -323,7 +324,16 @@ func (c *component) createUser(ctx context.Context, accessToken string, realmNam
 	kcUser.Groups = &groupIDs
 
 	// Set onboarding status
-	kcUser.SetAttributeString(constants.AttrbOnboardingStatus, registerOnboardingStatus)
+	realm := ctx.Value(cs.CtContextRealm).(string)
```

Comment:
Use the realmName parameter.
Not even sure the realmName is available in the context ofr the register.